OpenRA.Mods.Common/Widgets/Logic/ServerCreationLogic.cs

```diff
@@ -90,6 +90,10 @@ public ServerCreationLogic(Widget widget, ModData modData, Action onExit, Action
 			checkboxUPnP.OnClick = () => allowPortForward ^= true;
 			checkboxUPnP.IsDisabled = () => !Game.Settings.Server.AllowPortForward;
 
+			var labelUPnP = panel.GetOrNull<LabelWidget>("UPNP_NOTICE");
+			if (labelUPnP != null)
+				labelUPnP.IsVisible = () => !Game.Settings.Server.DiscoverNatDevices;
```

Comment:
`Game.Settings.Server.AllowPortForward` is a setting. It decides whether to do automatic port forwarding if a NAT device is available and let's the user decide whether he wants to or not. You may be confusing this with `NatDeviceAvailable` which indeed could be moved to UPnP or replaced by a `NatDevice == null` check.